### PR TITLE
Cucumber-TestNg Webdriver Integration

### PR DESCRIPTION
**Pull request by Rushabhshroff**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,112 @@
...
+
+  ```shell
+  set BROWSERSTACK_USERNAME=<browserstack-username>
+  set BROWSERSTACK_ACCESS_KEY=<browserstack-access-key>
+  ```
+- Upload your Android/iOS app to BrowserStack using the REST API as mentioned [here](https://www.browserstack.com/docs/app-automate/api-reference/appium/apps#upload-an-app).
+
+## Configuration
+
+This repository allows you to to use configuration files similar to the following to run your Appium tests on various platforms including on-premise devices, devices running on an on-prem grid or remote elenium grid.
```
Comment (samirans89) [inline README.md]: `remote elenium grid.` - Typo.
Do we want to mention Appium? 
Also do we want to call out that only cloud config has been implemented.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,112 @@
...
+
+Please make sure that the following keys are set in the ```capabilities.yml``` config file. 
+```yml
+driverType: cloudDriver
+mobileAppTesting: true
+cloudDriver:
+  app:
+    android_custom_id: <your_android_custom_id>
+```
+The above is just an example. You can use any of hash/custom/shareable ID for your Android/iOS apps.
```
Comment (samirans89) [inline README.md]: do we want to mention hash id OR custom id OR shareable ID. As in, my understanding is that only one of them is needed.

Comment (samirans89) [reply README.md]: also, which one is chosen if all the IDs are present?

File: pom.xml [hunk 3]
```diff
@@ -98,12 +116,12 @@
...
-      <artifactId>webdriver-framework-testng</artifactId>
-      <version>${webdriver-framework.version}</version>
-      <scope>test</scope>
-    </dependency>
+<!--    <dependency>-->
+<!--      <groupId>com.browserstack.webdriver</groupId>-->
+<!--      <artifactId>webdriver-framework-testng</artifactId>-->
+<!--      <version>${webdriver-framework.version}</version>-->
+<!--      <scope>test</scope>-->
+<!--    </dependency>-->
```
Comment (samirans89) [inline pom.xml]: Do we want to state - We can uncomment these lines once the maven dependency version with AA is uploaded in Maven or something similar

File: src/test/resources/capabilities.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    ios_path: /path/to/ipa
+    android_path: /path/to/apk
+  platforms:
+    - name:
+      os: Android,
+      os_version: 11.0
+      device: Google Pixel 4
+      capabilities:
+        appPackage: com.browserstack.sampeapp
+        appActivity: com.browserstack.sampeapp.MainActivity
```
Comment (samirans89) [inline src/test/resources/capabilities.yml]: If we do not have constraint on the app name and activity, can we update `sampeapp` to `sampleapp`

**Review by samirans89 — CHANGES_REQUESTED** [review]

File: src/test/resources/capabilities.yml (hunk 4 above)
Comment (arvind1998) [reply src/test/resources/capabilities.yml]: done

File: pom.xml (hunk 3 above)
Comment (arvind1998) [reply pom.xml]: this isn't required at all. have removed it altogether.

File: README.md (hunk 2 above)
Comment (Rushabhshroff) [reply README.md]: Hash Id is given preference over Custom ID or Sharable ID

Comment (arvind1998) [reply README.md]: modified

File: README.md (hunk 1 above)
Comment (arvind1998) [reply README.md]: typo fixed. 
haven't added the second part as we haven't tested it out.

**Review by anirudha13 — APPROVED** [review]

**Comment by arvind1998** [comment]
@samirans89 , can we merge this now?

**Review by samirans89 — APPROVED** [review]

**Pull request merged**